capstone/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -12,8 +18,29 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ReadData readData = new ReadData();
-    readData.readAll();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    ReadData readData = new ReadData(new SentimentData());
+    ArrayList<Entity> allEntities = readData.allEntitiesFromFiles();
+
+    for (int i = 0; i < allEntities.size(); i++) {
+      Entity currentEntity = allEntities.get(i);
+      Entity inStore = null;
+
+      try {
+        inStore = datastore.get(currentEntity.getKey());
+      } catch (EntityNotFoundException e) {
+        inStore = null;
```

Comment:
You don't need to set inStore again here, you can just continue the for loop by adding a `continue` and the rest of this logic won't run.

Reply:
This logic including the logic below.